macaroon-thought.md

```diff
@@ -0,0 +1,399 @@
+# Integral Principles of the Structural Dynamics of Macaroons
+
+## Describe a Fly Macaroon In The Fewest Words Possible
+
+A Macaroon is an HMAC-authenticated access token. Fly Macaroons are encoded 
+in MsgPack, which is a kind of binary JSON. Macaroons are like other kinds of access tokens, with these quirks:
+
+* They're cryptographically secured using a minimal, non-negotiable set of
+  primitives: HMAC and an AEAD cipher. There are no crypto parameters to set
+  or verify.
+  
+* They're rigidly structured: every Macaroon is a set of restrictions on 
+  what its holder can do. These are called caveats. All caveats 
+  must clear for the token to authorize an action. 
```

Comment:
You describe caveats needing to _clear_. That's a succinct way to describe a caveat permitting the user's access, but we don't define it anywhere and it's probably confusing. I think we should either use more words in place of the term _clear_ or we should define early on what we mean by it.